src/osd/PGLog.h

```diff
@@ -512,11 +512,6 @@ struct PGLog {
     if (missing.is_missing(oid, v)) {
       missing.got(oid, v);
       
-      // raise last_complete?
-      if (missing.missing.empty()) {
-	log.complete_to = log.log.end();
-	info.last_complete = info.last_update;
-      }
       while (log.complete_to != log.log.end()) {
 	if (missing.missing[missing.rmissing.begin()->second].need <=
 	    log.complete_to->version)
```

Comment:
With this change we run teh risk of (less efficiently) running through this loop to advance to the end of the log on the last recovered object.  Perhaps we should leave this hunk behind and just add the second (nearly duplicate) case